service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/utils/DocumentReferenceUtilsTest.java

```diff
@@ -1,16 +1,76 @@
 package uk.nhs.adaptors.gp2gp.ehr.utils;
 
+import org.apache.tika.mime.MimeTypeException;
 import org.hl7.fhir.dstu3.model.Attachment;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
 import uk.nhs.adaptors.gp2gp.ehr.exception.EhrMapperException;
 
+import java.util.AbstractMap;
+import java.util.Map;
+
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertAll;
 
 class DocumentReferenceUtilsTest {
     private static final String TEXT_PLAIN_CONTENT_TYPE = "text/plain";
     private static final String NARRATIVE_STATEMENT_ID = "3b24b89b-fd14-49f9-ba12-3b4212b60080";
     private static final String FILE_MISSING_ATTACHMENT_TITLE = "Reason why file is missing";
+    private static final Map<String, String> SUPPORTED_MIME_TYPES;
+    private static final String UNSUPPORTED_CONTENT_TYPE = "unknown/unknown";
+    private static final String INVALID_CONTENT_TYPE = "invalid content type";
+
+    static {
+        SUPPORTED_MIME_TYPES = Map.ofEntries(
+            new AbstractMap.SimpleEntry<>("text/plain", ".txt"),
+            new AbstractMap.SimpleEntry<>("text/html", ".html"),
+            new AbstractMap.SimpleEntry<>("application/pdf", ".pdf"),
+            new AbstractMap.SimpleEntry<>("text/xml", ".xml"),
+            new AbstractMap.SimpleEntry<>("application/xml", ".xml"),
+            new AbstractMap.SimpleEntry<>("text/rtf", ".rtf"),
+            new AbstractMap.SimpleEntry<>("audio/mpeg", ".mp3"),
+            new AbstractMap.SimpleEntry<>("image/png", ".png"),
+            new AbstractMap.SimpleEntry<>("image/gif", ".gif"),
+            new AbstractMap.SimpleEntry<>("image/jpeg", ".jpg"),
+            new AbstractMap.SimpleEntry<>("image/tiff", ".tiff"),
+            new AbstractMap.SimpleEntry<>("video/mpeg", ".mpeg"),
+            new AbstractMap.SimpleEntry<>("application/msword", ".doc"),
+            new AbstractMap.SimpleEntry<>("application/octet-stream", ".bin"),
+            new AbstractMap.SimpleEntry<>("audio/basic", ".au"),
+            new AbstractMap.SimpleEntry<>("audio/x-au", ".au")
```

Comment:
with `import static java.util.Map.entry` this is
```suggestion
            entry("text/plain", ".txt"),
            entry("text/html", ".html"),
            entry("application/pdf", ".pdf"),
            entry("text/xml", ".xml"),
            entry("application/xml", ".xml"),
            entry("text/rtf", ".rtf"),
            entry("audio/mpeg", ".mp3"),
            entry("image/png", ".png"),
            entry("image/gif", ".gif"),
            entry("image/jpeg", ".jpg"),
            entry("image/tiff", ".tiff"),
            entry("video/mpeg", ".mpeg"),
            entry("application/msword", ".doc"),
            entry("application/octet-stream", ".bin"),
            entry("audio/basic", ".au"),
            entry("audio/x-au", ".au")
```

Reply:
*BUT* I think this would work better just as a `@ParameterizedTest @CsvSource(...)`. What do you think?

Reply:
I have no idea why I haven't thought about that :-)